lib/CL/devices/printf_buffer.c

```diff
@@ -377,7 +371,7 @@ __pocl_printf_format_full (param_t *p, char *buffer, uint32_t buffer_size)
                             "printf error: vector-length is zero\n");
                           return -1;
                         }
-                      if (vector_length > (UINT32_MAX - 9) / 10)
+                      if (vector_length < 0)
```

Comment:
Signed integer overflow is undefined behaviour. In this case though, we can just change the condition to: `vector_length > 16` - there are no valid larger vector lengths (and AFAIK no extensions with larger vectors).

Reply:
fixed.